subprojects/instant-execution/src/main/kotlin/org/gradle/instantexecution/serialization/beans/BeanPropertyWriter.kt

```diff
@@ -41,22 +43,30 @@ class BeanPropertyWriter(
      * Serializes a bean by serializing the value of each of its fields.
      */
     override suspend fun WriteContext.writeStateOf(bean: Any) {
-        for (relevantField in relevantFields) {
-            val field = relevantField.field
-            val fieldName = field.name
-            val originalFieldValue = field.get(bean)
-            val fieldValue = originalFieldValue ?: conventionalValueOf(bean, fieldName)
-            relevantField.unsupportedFieldType?.let {
-                reportUnsupportedFieldType(it, "serialize", field.name, fieldValue)
+        withDebugFrame({ GeneratedSubclasses.unpackType(bean).name }) {
```

Comment:
I would consider instrumenting `BindingsBackedCodec.encode()` or `DefaultWriteContext.write(Any?)`  instead of individual codecs, so that we instrument everything. When I added instrumentation to see what was going on in the square mock-up, I found this very useful as often it's our hand-coded codecs that are a problem, or at least they were in this case. Without instrumenting everything the result would have been misleading and I would have missed certain problems.

Reply:
On one hand, instrumenting at the `BindingsBackedCodec` level will provide insight on the space used by primitives, collections and our internal codecs, on the other hand, it's going to be a lot of data and I feared information overload. 

In any case, let's give that a try and see what the experience is like.

Reply:
Yep, much better that way. 👍